src/SLCore.Listeners/Implementation/AnalysisListener.cs

```diff
@@ -29,18 +30,64 @@ namespace SonarLint.VisualStudio.SLCore.Listeners.Implementation;
 [PartCreationPolicy(CreationPolicy.Shared)]
 internal class AnalysisListener : IAnalysisListener
 {
+    private readonly IEnumerable<Language> supportedLanguages;
+
+    private readonly IAnalysisService analysisService;
+    private readonly IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter;
+    private readonly IAnalysisStatusNotifierFactory analysisStatusNotifierFactory;
+
+    [ImportingConstructor]
+    public AnalysisListener(IAnalysisService analysisService, IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter, IAnalysisStatusNotifierFactory analysisStatusNotifierFactory)
+       : this(analysisService, raiseIssueParamsToAnalysisIssueConverter, analysisStatusNotifierFactory, new List<Language>() { Language.Secrets })
+    {
+    }
+
+    internal AnalysisListener(IAnalysisService analysisService, IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter, IAnalysisStatusNotifierFactory analysisStatusNotifierFactory, IEnumerable<Language> supportedLanguages)
+    {
+        this.supportedLanguages = supportedLanguages;
+        this.analysisService = analysisService;
+        this.raiseIssueParamsToAnalysisIssueConverter = raiseIssueParamsToAnalysisIssueConverter;
+        this.analysisStatusNotifierFactory = analysisStatusNotifierFactory;
+    }
+
     public Task DidChangeAnalysisReadinessAsync(DidChangeAnalysisReadinessParams parameters)
     {
         return Task.CompletedTask;
     }
 
     public void RaiseIssues(RaiseIssuesParams parameters)
     {
-        throw new NotImplementedException();
+        if (parameters.analysisId.HasValue)
+        {
+            var filepath = parameters.issuesByFileUri.Single().Key.LocalPath;
+
+            var issues = GetSupportedLanguageIssues(raiseIssueParamsToAnalysisIssueConverter.GetAnalysisIssues(parameters));
+
+            if (issues.Any())
+            {
+                analysisService.PublishIssues(filepath, parameters.analysisId.Value, issues);
+            }
+            if (!parameters.isIntermediatePublication)
+            {
+                var analysisStatusNotifier = analysisStatusNotifierFactory.Create("SLCoreAnalyzer", filepath);
```

Comment:
this magic string will need to be changed after analyzer task is done with `nameof`

Reply:
yep